[easy] Fix most documentation warnings

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27782 [easy] Fix most documentation warnings**

Warnings show up when running `make html` to build documentation. All of
the warnings are very reasonable and point to bugs in our docs. This PR
attempts to fix most of those warnings.

In the future we will add something to the CI that asserts that there
are no warnings in our docs.

Test Plan:
- build and view changes locally

Differential Revision: [D17887067](https://our.internmc.facebook.com/intern/diff/D17887067)

zou3519 committed:
Fix most documentation warnings

Warnings show up when running `make html` to build documentation. All of
the warnings are very reasonable and point to bugs in our docs. This PR
attempts to fix most of those warnings.

In the future we will add something to the CI that asserts that there
are no warnings in our docs.

Test Plan:
- build and view changes locally

zou3519 committed:
Update on "[easy] Fix most documentation warnings"


Warnings show up when running `make html` to build documentation. All of
the warnings are very reasonable and point to bugs in our docs. This PR
attempts to fix most of those warnings.

In the future we will add something to the CI that asserts that there
are no warnings in our docs.

Test Plan:
- build and view changes locally

zou3519 committed:
Update on "[easy] Fix most documentation warnings"


Warnings show up when running `make html` to build documentation. All of
the warnings are very reasonable and point to bugs in our docs. This PR
attempts to fix most of those warnings.

In the future we will add something to the CI that asserts that there
are no warnings in our docs.

Test Plan:
- build and view changes locally

Differential Revision: [D17887067](https://our.internmc.facebook.com/intern/diff/D17887067)

jlin27 commented on docs/source/packages.rst:
```diff
@@ -1,3 +1,5 @@
+:orphan:
```
This rst file is needed for Javadocs to appear in the toc. I think there's a formatting issue in the index.rst of master that's preventing packages.rst to be picked up. Can we remove this orphan directive?

zou3519 committed:
Update on "[easy] Fix most documentation warnings"


Warnings show up when running `make html` to build documentation. All of
the warnings are very reasonable and point to bugs in our docs. This PR
attempts to fix most of those warnings.

In the future we will add something to the CI that asserts that there
are no warnings in our docs.

Test Plan:
- build and view changes locally

Differential Revision: [D17887067](https://our.internmc.facebook.com/intern/diff/D17887067)
